### DISCUSSION TITLE
Error 412 precondition_failed

### DISCUSSION THREAD

**Discussion by SylvainGa** [teslamate-org/teslamate#3647] [General]
Some users of an App I maintain for a Garmin watch are starting to get Error 412 (precondition_failed) when authenticating to their vehicle According to a Google search this error is:

The HyperText Transfer Protocol (HTTP) 412 Precondition Failed client error response code indicates that access to the target resource has been denied. 

I just checked my Teslamate instance and the drives I did today were recorded so looks like either I'm not affected yet or Teslamate won't be, however if it's the Auth process that is barking, Teslamate will also be affected for most.

**Comment by cwanja** [comment]
What it sounds like is actually being impacted is the **commands** API. Would suggest you take a look through here:
https://www.tesla.com/developer-docs, specifically here: https://developer.tesla.com/docs/fleet-api#vehicle-commands

Edit: more details https://developer.tesla.com/docs/fleet-api#2023-10-09-rest-api-vehicle-commands-endpoint-deprecation-warning

**Reply by SylvainGa** [reply]
I don't think it's the command API here because the error is received when the app launches, before any commands can be sent. The order of the process is:

1. Check if the access token is still valid, if not, get a new access token from the refresh one
2. Authenticate with the access token
3. Ask for vehicle data via the GET /api/1/vehicles/{id}/vehicle_data URL
4. If the command returns 408, see if awake by using the GET /api/1/vehicles URL. If not, ask the user if ok to wake. 

Nowhere there am I sending a command to the vehicle. It's just queries, beside the authentication.

**Reply by cwanja** [reply]
Could be related to #3630 based upon #3629
